src/core/namespace.c

```diff
@@ -1190,6 +1190,66 @@ static void normalize_mounts(const char *root_directory, MountEntry *mounts, siz
         drop_nop(mounts, n_mounts);
 }
 
+static bool root_read_only(
+                char **read_only_paths,
+                ProtectSystem protect_system) {
+
+        /* Determine whether the root directory is going to be read-only given the configured settings. */
+
+        if (protect_system == PROTECT_SYSTEM_STRICT)
+                return true;
+
+        if (path_strv_contains(read_only_paths, "/"))
+                return true;
+
+        return false;
+}
+
+static bool home_read_only(
+                char** read_only_paths,
+                char** inaccessible_paths,
+                char** empty_directories,
+                const BindMount *bind_mounts,
+                size_t n_bind_mounts,
+                const TemporaryFileSystem *temporary_filesystems,
+                size_t n_temporary_filesystems,
+                ProtectHome protect_home) {
+
+        size_t i;
+
+        /* Determine whether the /home directory is going to be read-only given the configured settings. Yes,
+         * this is a bit sloppy, since we don't bother checking for cases where / is affected by multiple
+         * settings. */
+
+        if (protect_home != PROTECT_HOME_NO)
+                return true;
+
+        if (path_strv_contains(read_only_paths, "/home") ||
+            path_strv_contains(inaccessible_paths, "/home") ||
+            path_strv_contains(empty_directories, "/home"))
+                return true;
+
+        for (i = 0; i < n_temporary_filesystems; i++)
+                if (path_equal(temporary_filesystems[i].path, "/home"))
+                        return true;
+
+        /* We consider /home read-only if either there's a read-only bind mount on /home or there's a
+         * writable one from a source outside of /home. */
+
+        for (i = 0; i < n_bind_mounts; i++) {
+                if (!path_equal(bind_mounts[i].destination, "/home"))
+                        continue;
+
+                if (bind_mounts[i].read_only)
+                        return true;
+
+                if (!path_startswith(bind_mounts[i].source, "/home"))
```

Comment:
I'm doing `BindPaths=/home:/home`, so the root is still read-write even if /home is a different partition. How about looking at the block devices for /home and root. If they are different, mounting root as read-only will not affect /home?

Reply:
ah, eh, i forgot that BindPaths= only binds stuff from the host, never from the container... hence it's sufficient to check the destination, the source doesn't need to be checked and read-only or not doesn't need to be checked either. will prep fix.